flowzone.yml

```diff
@@ -488,13 +488,20 @@ jobs:
       - name: Check for package.json
         id: npm
         run: |
+          runner_os="$(echo "${RUNNER_OS}" | tr '[:upper:]' '[:lower:]')"
           if test -f "package.json"
           then
             echo "found package.json"
-            echo "enabled=true" >> $GITHUB_OUTPUT
-            echo "private=$(jq -r '.private' package.json)" >> $GITHUB_OUTPUT
-            echo "docs=$(jq -r '.scripts | has("doc")' package.json)" >> $GITHUB_OUTPUT
-            echo "NODE_VERSIONS=[]" >> $GITHUB_ENV
+            os="$(cat < package.json | yq e .os[])"
```

Comment:
> it only makes sense to compare runner_os with os key for value linux. That's the only convergence, between the two sets of values. There are more verbose way of checking, or we can just limit the npm steps to run if: runner_os == 'Linux'.

Not sure if you accidentally deleted this comment, but I'm not sure I follow.

In package.json `os` is an array of strings. It could be `["windows"]` or `["linux"]` or `["windows", "linux", "macos"]`. If ANY of them match the current runner OS (only linux for now, but eventually more) we would want to run the npm scripts.

Reply:
@klutchell I think we are good here, right? Current logic runs npm workflow if either the `os` key is omitted in `package.json` (i.e. most cases), or exists and matches `runner_os`.

Reply:
Oh I see, you were checking for an empty array, which I forgot to do.